lib/node_modules/@stdlib/repl/lib/tokenizer.js

```diff
@@ -0,0 +1,329 @@
+	VISITORS = {
+		'Identifier': Identifier,
+		'MemberExpression': MemberExpression
+	};
+	walk.simple( ast, VISITORS );
+
+	return tokens;
+
+	/**
+	* Callback invoked upon encountering a `Token` when parsing.
+	*
+	* @private
+	* @param {Object} token - token object
+	*/
+	function onToken( token ) {
+		// Ignore non-existent tokens:
+		if ( token.start === token.end ) {
+			return;
+		}
+		if ( token.type.isLoop || isControlKeyword( token.type.keyword ) ) {
+			// Control flow keywords - `for`, `while`, `do`, `if`, `else` etc:
+			token.type = 'control';
+			tokens.push( token );
+			return;
+		}
+		if ( isSpecialIdentifier( token.type.keyword ) ) {
+			// Special identifiers - `this`, `super`:
+			token.type = 'specialIdentifier';
+			tokens.push( token );
+			return;
+		}
+		if ( isReservedLiteral( token.type.keyword ) || ( token.type.label === 'name' && token.value === 'undefined' ) ) {
+			// Built-in literals - `true`, `false`, `null`, `undefined`:
+			token.type = 'literal';
+			tokens.push( token );
+			return;
+		}
+		if ( token.type.keyword || ( token.type.label === 'name' && isUnrecognizedKeyword( token.value ) ) ) {
+			// Keywords - `function`, `import`, `var`, `const`, `let` etc.:
+			token.type = 'keyword';
+			tokens.push( token );
+			return;
+		}
+		if ( isStringTokenType( token.type.label ) ) {
+			// Strings and template string literals:
+			token.type = 'string';
+			tokens.push( token );
+			return;
+		}
+		if ( token.type.label === 'regexp' ) {
+			// Regex expressions:
+			token.type = 'regexp';
+			tokens.push( token );
+			return;
+		}
+		if ( token.type.label === 'num' ) {
+			// Numeric literals:
+			token.type = 'number';
+			tokens.push( token );
+			return;
+		}
+		if ( token.type.binop || token.type.prefix || token.type.postfix || token.type.isAssign ) { // eslint-disable-line max-len
+			// Operator symbols - `+`, `=`, `++` etc:
+			token.type = 'operator';
+			tokens.push( token );
+			return;
+		}
+		if ( RE_PUNCTUATION.test( token.type.label ) ) {
+			// Punctuation symbols - `,`, `(`, `;` etc:
+			token.value = token.type.label;
+			token.type = 'punctuation';
+			tokens.push( token );
+		}
+	}
+
+	/**
+	* Callback invoked upon encountering a `Comment` when parsing.
+	*
+	* @private
+	* @param {Object} block - true if the comment is a block comment, false if it is a line comment
+	* @param {string} text - comment value
+	* @param {number} start - start index
+	* @param {number} end - end index
+	*/
+	function onComment( block, text, start, end ) {
+		if ( block ) {
+			// TODO: add support for highlighting multi-line comments.
+			return;
+		}
+		tokens.push({
+			'type': 'comment',
+			'start': start,
+			'end': end
+		});
+	}
+
+	/**
+	* Callback invoked upon encountering a `Identifier` AST node.
+	*
+	* @private
+	* @param {Object} node - AST node
+	*/
+	function Identifier( node ) {
+		var identifier;
+		var command;
+		var i;
+
+		// Ignore non-existent nodes:
```

Comment:
What does it mean for a node to be "non-existent"?

Reply:
The nodes that acorn-loose adds by itself, X symbol

Reply:
This would be good to explain in the comment, as otherwise the reader lacks context for why this would be applicable.

Reply:
It also includes eof nodes, yes I should mention that